judo/controller/controller.py

```diff
@@ -241,6 +262,25 @@ def update_traces(self) -> None:
             elites[sensor :: self.num_trace_sensors] = s1
         self.traces = np.reshape(elites, (total_traces_rollouts, 2, 3))
 
+    def _init_action_normalizer(self) -> Normalizer:
+        """Initialize the action normalizer."""
+        action_normalizer_kwargs = {}
+        if self.action_normalizer_type == "min_max":
+            action_normalizer_kwargs["min"] = self.task.actuator_ctrlrange[:, 0]
+            action_normalizer_kwargs["max"] = self.task.actuator_ctrlrange[:, 1]
+        elif self.action_normalizer_type == "running_mean_std":
+            action_normalizer_kwargs["init_std"] = 1.0
+        return make_normalizer(self.action_normalizer_type, self.model.nu, **action_normalizer_kwargs)
+
+    def update_action_normalizer(self, actions: np.ndarray, rewards: np.ndarray) -> None:
+        """Update the action normalizer."""
+        if normalizer_registry.get(self.action_normalizer_type) != type(self.action_normalizer):
+            # Force the normalizer to be re-initialized when the type changes in GUI
+            self.action_normalizer = self._init_action_normalizer()
+
+        # Use top 20% of actions to update the normalizer
+        self.action_normalizer.update(actions[rewards.argsort()[-int(0.2 * len(rewards)) :]])
```

Comment:
three comments:
1. the 0.2 ratio is a magic number and should be exposed somewhere if you're going to use it
2. i don't think that the normalizer should only be updated based on the top percentage of samples, because the decision that the optimizer is making is also over non-performant samples
3. i think all the logic for updating the action normalizer should just be in the `update` function of the normalizer anyway (this only affects the running normalizer)